api/views/phones.py

```diff
@@ -611,6 +613,87 @@ def sms_status(request):
     return response.Response(status=200)
 
 
+call_body = openapi.Schema(
+    type=openapi.TYPE_OBJECT,
+    required=["to"],
+    properties={"to": openapi.Schema(type=openapi.TYPE_STRING)},
+)
+
+
+@decorators.permission_classes([permissions.IsAuthenticated, HasPhoneService])
+@swagger_auto_schema(method="post", request_body=call_body)
+@decorators.api_view(["POST"])
+def call(request):
+    """
+    Make a call from the authenticated user's relay number.
+
+    """
+    to = request.data.get("to", None)
+    if to is None:
+        raise exceptions.ValidationError("Missing 'to' parameter.")
+    real_phone = RealPhone.objects.get(user=request.user)
+    relay_number = RelayNumber.objects.get(user=request.user)
+    client = twilio_client()
+    client.calls.create(
+        twiml=f"<Response><Say>Dialing {to} ...</Say><Dial>{to}</Dial></Response>",
+        to=real_phone.number,
+        from_=relay_number.number,
+    )
+    return response.Response(status=200)
+
+
+messages_body = openapi.Schema(
+    type=openapi.TYPE_OBJECT,
+    properties={"with": openapi.Schema(type=openapi.TYPE_STRING)},
+)
+
+
+@decorators.permission_classes([permissions.IsAuthenticated, HasPhoneService])
+@decorators.api_view(["GET"])
+def messages(request):
+    """
+    Get the user's messages.
+
+    Pass ?with=<E.164> parameter to filter the messages to only the ones sent between
+    the relay number and the <E.164> number.
+
+    """
+    _with = request.data.get("with", None)
+    relay_number = RelayNumber.objects.get(user=request.user)
+
+    contact = None
+    if _with is not None:
+        contact = InboundContact.objects.get(
+            relay_number=relay_number, inbound_number=_with
+        )
+
+    client = twilio_client()
+    if contact:
+        inbound_messages = client.messages.list(
+            from_=contact.inbound_number, to=relay_number.number
+        )
+        outbound_messages = client.messages.list(
+            from_=relay_number.number, to=contact.inbound_number
+        )
+        return response.Response(
+            {
+                "inbound_messages": inbound_messages,
+                "outbound_messages": outbound_messages,
+            },
+            status=200,
```

Comment:
Could you split the messages into two parts instead of having it contain both `inbound_messages` and `outbound_messages`?

Reply:
How about a url parameter?

Reply:
Yeah that works too